packages/abi-typegen/src/abi/types/EnumType.ts

```diff
@@ -40,6 +40,29 @@ export class EnumType extends AType implements IType {
     return name;
   }
 
+  public getNativeEnum(params: { types: IType[] }) {
+    const { types } = params;
+
+    const typeHash: { [key: number]: IType['rawAbiType'] } = types.reduce(
+      (hash, row) => ({
+        ...hash,
+        [row.rawAbiType.typeId]: row,
+      }),
+      {}
+    );
+
+    const { components } = this.rawAbiType;
+
+    // `components` array guaranteed to always exist for structs/enums
+    const enumComponents = components as IRawAbiTypeComponent[];
+
+    if (!enumComponents.every(({ type }) => !typeHash[type])) {
+      return undefined;
+    }
```

Comment:
In my [previous](https://github.com/FuelLabs/fuels-ts/pull/880#discussion_r1160175034) review, I suggested using the `findType` helper instead of adding a duplicate called `typeHash`.

Is anything wrong with that?

```suggestion
    if (!enumComponents.every(({ type: typeId }) => !findType({ types, typeId })) {
      return undefined;
    }
```

Reply:
Make sense. I will give this a complete review soon.